In [1]:
from google.colab import drive
drive.mount('/gdrive', force_remount=True)

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /gdrive


In [0]:
import pandas as pd
plant1_train = pd.read_csv("/gdrive/My Drive/Plain/plant1_train.csv")
plant2_train = pd.read_csv("/gdrive/My Drive/Plain/plant2_train.csv")

In [3]:
plant1_train = plant1_train.drop('Unnamed: 0' , axis=1)
cols = ['Date' , 'loc1_tem', 'loc1_hum', 'loc1_coil_temp','loc2_tem', 'loc2_hum', 'loc2_coil_temp' , 'loc3_tem', 'loc3_hum', 'loc3_coil_temp', 'out_tem', 'out_hum', 'loc1' , 'loc2' , 'loc3']
plant1_train.columns = cols
plant1_train_first = plant1_train[['Date','loc1_tem', 'out_tem' , 'loc1_coil_temp','loc1_hum' ,'out_hum','loc1']]
plant1_train_first["Hour"] = pd.to_datetime(plant1_train_first["Date"])
plant1_train_first["Hour"] = plant1_train_first["Hour"].apply(lambda x: x.hour)
tmp = []
for i in plant1_train_first["Hour"]:
  if 0<=i<6:
    tmp.append(1)
  elif 6<=i<12:
    tmp.append(2)
  elif 12<= i < 18:
    tmp.append(3)
  else:
    tmp.append(4)
plant1_train_first["Hour_label"] = tmp
plant1_train_first

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-do

Date  loc1_tem  out_tem  ...  loc1  Hour  Hour_label
0       2016-04-01 0:00     16.00     9.00  ...   0.0     0           1
1       2016-04-01 3:00     14.00     7.00  ...   0.0     3           1
2       2016-04-01 6:00     13.00     6.00  ...   0.0     6           2
3       2016-04-01 9:00     13.00    18.00  ...   0.0     9           2
4      2016-04-01 12:00     16.00    18.00  ...   0.0    12           3
...                 ...       ...      ...  ...   ...   ...         ...
58744  2019-03-31 23:10     10.52     6.40  ...   0.0    23           4
58745  2019-03-31 23:20     10.52     6.34  ...   0.0    23           4
58746  2019-03-31 23:30     10.43     6.40  ...   0.0    23           4
58747  2019-03-31 23:40     10.49     6.31  ...   0.0    23           4
58748  2019-03-31 23:50     10.49     6.34  ...   0.0    23           4

[58749 rows x 9 columns]

In [4]:
plant1_train_second = plant1_train[['Date','loc2_tem', 'out_tem' , 'loc2_coil_temp','loc2_hum' ,'loc2']]
plant1_train_second["Hour"] = pd.to_datetime(plant1_train_second["Date"])
plant1_train_second["Hour"] = plant1_train_second["Hour"].apply(lambda x: x.hour)
tmp = []
for i in plant1_train_second["Hour"]:
  if 0<=i<6:
    tmp.append(1)
  elif 6<=i<12:
    tmp.append(2)
  elif 12<= i < 18:
    tmp.append(3)
  else:
    tmp.append(4)
plant1_train_second["Hour_label"] = tmp
plant1_train_second

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instea

Date  loc2_tem  out_tem  ...  loc2  Hour  Hour_label
0       2016-04-01 0:00     14.00     9.00  ...   0.0     0           1
1       2016-04-01 3:00     12.00     7.00  ...   0.0     3           1
2       2016-04-01 6:00     11.00     6.00  ...   0.0     6           2
3       2016-04-01 9:00     11.00    18.00  ...   0.0     9           2
4      2016-04-01 12:00     15.00    18.00  ...   0.0    12           3
...                 ...       ...      ...  ...   ...   ...         ...
58744  2019-03-31 23:10     10.55     6.40  ...   0.0    23           4
58745  2019-03-31 23:20     10.55     6.34  ...   0.0    23           4
58746  2019-03-31 23:30     10.49     6.40  ...   0.0    23           4
58747  2019-03-31 23:40     10.55     6.31  ...   0.0    23           4
58748  2019-03-31 23:50     10.55     6.34  ...   0.0    23           4

[58749 rows x 8 columns]

In [5]:
plant1_train_third = plant1_train[['Date','loc3_tem', 'out_tem' , 'loc3_coil_temp','loc3_hum' ,'loc3']]
plant1_train_third["Hour"] = pd.to_datetime(plant1_train_third["Date"])
plant1_train_third["Hour"] = plant1_train_third["Hour"].apply(lambda x: x.hour)
tmp = []
for i in plant1_train_third["Hour"]:
  if 0<=i<6:
    tmp.append(1)
  elif 6<=i<12:
    tmp.append(2)
  elif 12<= i < 18:
    tmp.append(3)
  else:
    tmp.append(4)
plant1_train_third["Hour_label"] = tmp
plant1_train_third

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instea

Date  loc3_tem  out_tem  ...  loc3  Hour  Hour_label
0       2016-04-01 0:00     13.00     9.00  ...   0.0     0           1
1       2016-04-01 3:00     11.00     7.00  ...   0.0     3           1
2       2016-04-01 6:00     10.00     6.00  ...   0.0     6           2
3       2016-04-01 9:00     10.00    18.00  ...   0.0     9           2
4      2016-04-01 12:00     14.00    18.00  ...   0.0    12           3
...                 ...       ...      ...  ...   ...   ...         ...
58744  2019-03-31 23:10      9.85     6.40  ...   0.0    23           4
58745  2019-03-31 23:20      9.88     6.34  ...   0.0    23           4
58746  2019-03-31 23:30      9.85     6.40  ...   0.0    23           4
58747  2019-03-31 23:40      9.82     6.31  ...   0.0    23           4
58748  2019-03-31 23:50      9.82     6.34  ...   0.0    23           4

[58749 rows x 8 columns]

요기까지가 전처리

In [6]:
#tmp = plant1_train_first[['loc1' , 'loc1_tem' , 'out_tem', 'loc1_coil_temp', 'loc1_hum' , 'Hour_label']]
plant1_train_first = plant1_train_first.dropna(axis=0)
tmp = plant1_train_first.loc[:,"loc1_tem" : "Hour_label"]
print(len(tmp))



X =tmp.drop('loc1' , axis=1)
y = plant1_train_first['loc1']

from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score  , confusion_matrix
lr = LogisticRegression(random_state = 123 , solver = 'sag')
model = lr.fit(X = X,y = y)
acc = model.score(X,y)

print(acc)

57879
0.9952659859361772


/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/_sag.py:330: ConvergenceWarning: The max_iter was reached which means the coef_ did not converge
  "the coef_ did not converge", ConvergenceWarning)


In [0]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score  , confusion_matrix
lr = LogisticRegression(random_state = 123 , solver = 'sag')

acc = model.score(X,y)

In [27]:
plant1_train_first.corr()

loc1_tem   out_tem  ...      Hour  Hour_label
loc1_tem        1.000000  0.975129  ...  0.064765    0.068963
out_tem         0.975129  1.000000  ...  0.088736    0.089008
loc1_coil_temp  0.992906  0.958871  ...  0.018636    0.021802
loc1_hum        0.441913  0.447675  ... -0.141039   -0.143200
out_hum         0.227791  0.122572  ... -0.216506   -0.210765
loc1           -0.024984 -0.002416  ...  0.007465    0.007107
Hour            0.064765  0.088736  ...  1.000000    0.968979
Hour_label      0.068963  0.089008  ...  0.968979    1.000000

[8 rows x 8 columns]

24시간으로 전처리

In [8]:
tmp1 = plant1_train_first[plant1_train_first['Hour']==0]

tmp1['After24'] = tmp1['loc1'].shift(-1)
tmp1[tmp1['After24']==1]

tmp_loc1 = tmp1[tmp1['After24']==1]

for i in range(1,24):
  tmp1 = plant1_train_first[plant1_train_first['Hour']==i]

  tmp1['After24'] = tmp1['loc1'].shift(-1)
  tmp1 = tmp1[tmp1['After24']==1]  
  tmp_loc1 = pd.concat([tmp_loc1 , tmp1])

tmp_loc1

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  # This is added back by InteractiveShellApp.init_path()


Date  loc1_tem  out_tem  ...  Hour  Hour_label  After24
112     2016-04-15 0:00     16.00     7.00  ...     0           1      1.0
600     2016-06-15 0:00     25.00    22.00  ...     0           1      1.0
2080    2016-12-21 0:00     10.67     5.36  ...     0           1      1.0
3348    2017-02-16 0:00      6.83     1.12  ...     0           1      1.0
3420    2017-02-19 0:00      4.41    -2.09  ...     0           1      1.0
...                 ...       ...      ...  ...   ...         ...      ...
41893  2018-12-03 23:00     13.21    11.59  ...    23           4      1.0
41894  2018-12-03 23:10     13.08    11.59  ...    23           4      1.0
41895  2018-12-03 23:20     13.05    11.53  ...    23           4      1.0
41896  2018-12-03 23:30     13.05    11.53  ...    23           4      1.0
41897  2018-12-03 23:40     12.99    11.53  ...    23           4      1.0

[275 rows x 10 columns]

In [9]:
#tmp_loc1.drop(['Date', 'out_tem','loc1_coil_temp' , 'Hour_label'], axis=1).corr()
tmp_loc1 = tmp_loc1.drop(['Date', 'out_tem','loc1_coil_temp' , 'Hour_label'], axis=1)
tmp_loc1

loc1_tem  loc1_hum  out_hum  loc1  Hour  After24
112       16.00     50.00    84.00   0.0     0      1.0
600       25.00     68.00    68.00   0.0     0      1.0
2080      10.67     61.23    88.86   0.0     0      1.0
3348       6.83     41.30    57.42   0.0     0      1.0
3420       4.41     45.91    68.13   0.0     0      1.0
...         ...       ...      ...   ...   ...      ...
41893     13.21     85.99    88.83   1.0    23      1.0
41894     13.08     85.71    88.43   1.0    23      1.0
41895     13.05     85.50    88.22   1.0    23      1.0
41896     13.05     85.38    88.22   1.0    23      1.0
41897     12.99     85.26    88.25   1.0    23      1.0

[275 rows x 6 columns]

In [10]:
tmp_loc1_not = plant1_train_first[plant1_train_first['loc1']==0].sample(500)
tmp_loc1_not = tmp_loc1_not.drop(['Date', 'out_tem','loc1_coil_temp' , 'Hour_label'], axis=1)
tmp_loc1_not['After24'] = 0.0
tmp_loc1_not

loc1_tem  loc1_hum  out_hum  loc1  Hour  After24
46894      5.91     31.87    37.91   0.0    16      0.0
47175      4.35     49.66    73.60   0.0    15      0.0
42248      7.68     54.55    60.59   0.0    10      0.0
9298      17.08     58.73    57.33   0.0     8      0.0
53025      7.44     52.75    65.05   0.0     6      0.0
...         ...       ...      ...   ...   ...      ...
40156     11.37     50.40    75.95   0.0    21      0.0
22228     28.99     65.38    63.19   0.0     8      0.0
32263     23.83     43.89    56.11   0.0     0      0.0
36697     17.51     44.93    48.60   0.0    19      0.0
24479     32.62     64.26    79.12   0.0    23      0.0

[500 rows x 6 columns]

In [11]:
answer = pd.concat([tmp_loc1 , tmp_loc1_not] , axis= 0)
answer

loc1_tem  loc1_hum  out_hum  loc1  Hour  After24
112       16.00     50.00    84.00   0.0     0      1.0
600       25.00     68.00    68.00   0.0     0      1.0
2080      10.67     61.23    88.86   0.0     0      1.0
3348       6.83     41.30    57.42   0.0     0      1.0
3420       4.41     45.91    68.13   0.0     0      1.0
...         ...       ...      ...   ...   ...      ...
40156     11.37     50.40    75.95   0.0    21      0.0
22228     28.99     65.38    63.19   0.0     8      0.0
32263     23.83     43.89    56.11   0.0     0      0.0
36697     17.51     44.93    48.60   0.0    19      0.0
24479     32.62     64.26    79.12   0.0    23      0.0

[775 rows x 6 columns]

answer = plant1의 loc1에 대한 결과를 추출 (24시간 이후로)

Logistic 분류 99.5% (그날의 날씨) -> 85% (24시간 이후 날씨)

In [23]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score  , confusion_matrix

#x_cols = ['loc1_tem', 'loc1_hum' , 'out_hum' , 'loc1' , 'Hour']
x_cols = ['loc1_tem', 'loc1_hum' , 'out_hum']
y_cols = ['After24']

lr = LogisticRegression()
model = lr.fit(X = answer[x_cols],y = answer[y_cols])
acc = model.score(answer[x_cols],answer[y_cols])
acc

/usr/local/lib/python3.6/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


0.8516129032258064

In [13]:
x_cols = ['loc1_tem', 'loc1_hum' , 'out_hum' , 'loc1']
y_cols = ['After24']

lr = LogisticRegression()
model = lr.fit(X = answer[x_cols],y = answer[y_cols])
acc = model.score(answer[x_cols],answer[y_cols])
acc

/usr/local/lib/python3.6/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


0.8554838709677419

In [14]:
y_pred = model.predict(answer[x_cols])
confusion_matrix(answer[y_cols], y_pred)

array([[474,  26],
       [ 86, 189]])

SVC - 98.4

SVM의 시간복잡도는 O(n^3)

In [16]:
from sklearn.svm import SVC
#params = [0.001 , 0.01, 0.1 , 1 ,10, 100] # 10^-3 ~ 10^2
#kernel = ["rbf" , "linear" , "poly"]
svc = SVC(gamma = 'auto')

model = svc.fit( answer[x_cols] , answer[y_cols])
y_pred = model.predict(answer[x_cols])
print(confusion_matrix(answer[y_cols], y_pred))

acc = model.score(answer[x_cols],answer[y_cols])
print(acc)

[[498   2]
 [ 10 265]]
0.984516129032258


/usr/local/lib/python3.6/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [21]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.tree.export import export_text 

dtc = DecisionTreeClassifier(criterion = "gini", max_depth=2)
x_cols = ['loc1_tem', 'loc1_hum', 'out_hum']
model = dtc.fit( answer[x_cols] , answer[y_cols] )
print(export_text(model , feature_names = x_cols))

y_pred = model.predict(answer[x_cols])
print(confusion_matrix(answer[y_cols], y_pred))

acc = model.score(answer[x_cols],answer[y_cols])
print(acc)

|--- loc1_hum <= 76.42
|   |--- loc1_tem <= 15.01
|   |   |--- class: 0.0
|   |--- loc1_tem >  15.01
|   |   |--- class: 0.0
|--- loc1_hum >  76.42
|   |--- loc1_tem <= 24.97
|   |   |--- class: 1.0
|   |--- loc1_tem >  24.97
|   |   |--- class: 0.0

[[499   1]
 [133 142]]
0.8270967741935484


앙상블 XGBoost

In [22]:
from xgboost import XGBClassifier
model = XGBClassifier()
model.fit(X=answer[x_cols], y=answer[y_cols]) 
# n_estimators=100 : 트리를 100개 사용
# min_child_weight=1 : 노드를 분할할때 사용한 설명변수 개수

y_pred = model.predict(answer[x_cols])
print(confusion_matrix(answer[y_cols], y_pred))

acc = model.score(answer[x_cols],answer[y_cols])
print(acc)

[[488  12]
 [ 45 230]]
0.9264516129032258


/usr/local/lib/python3.6/dist-packages/sklearn/preprocessing/_label.py:235: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
/usr/local/lib/python3.6/dist-packages/sklearn/preprocessing/_label.py:268: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [26]:
plant1_train_second
tmp2 = plant1_train_second[plant1_train_second['Hour']==0]

tmp2['After24'] = tmp2['loc2'].shift(-1)
tmp2[tmp2['After24']==1]

tmp_loc2 = tmp2[tmp2['After24']==1]

for i in range(1,24):
  tmp2 = plant1_train_second[plant1_train_second['Hour']==i]

  tmp2['After24'] = tmp2['loc2'].shift(-1)
  tmp2 = tmp2[tmp2['After24']==1]  
  tmp_loc2 = pd.concat([tmp_loc2 , tmp2])

tmp_loc2

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  if sys.path[0] == '':


Date  loc2_tem  out_tem  ...  Hour  Hour_label  After24
112     2016-04-15 0:00     15.00     7.00  ...     0           1      1.0
2080    2016-12-21 0:00      8.87     5.36  ...     0           1      1.0
2850    2017-01-26 0:00     -0.10    -5.08  ...     0           1      1.0
3348    2017-02-16 0:00      4.60     1.12  ...     0           1      1.0
3420    2017-02-19 0:00      3.47    -2.09  ...     0           1      1.0
...                 ...       ...      ...  ...   ...         ...      ...
41893  2018-12-03 23:00     13.33    11.59  ...    23           4      1.0
41894  2018-12-03 23:10     13.24    11.59  ...    23           4      1.0
41895  2018-12-03 23:20     13.21    11.53  ...    23           4      1.0
41896  2018-12-03 23:30     13.14    11.53  ...    23           4      1.0
41897  2018-12-03 23:40     13.14    11.53  ...    23           4      1.0

[487 rows x 9 columns]

In [46]:
from sklearn.svm import SVC
#params = [0.001 , 0.01, 0.1 , 1 ,10, 100] # 10^-3 ~ 10^2
#kernel = ["rbf" , "linear" , "poly"]
svc = SVC(gamma = 'auto')

model = svc.fit( answer[x_cols] , answer[y_cols])
y_pred = model.predict(answer[x_cols])
print(confusion_matrix(answer[y_cols], y_pred))

acc = model.score(answer[x_cols],answer[y_cols])
print(acc)

[[498   2]
 [ 10 265]]
0.984516129032258


/usr/local/lib/python3.6/dist-packages/sklearn/utils/validation.py:760: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


In [50]:
tmp1 = plant1_train_first
tmp1['After24'] = tmp1['loc1'].shift(-1)

y_pred = model.predict(tmp1[x_cols][:-1])
print(confusion_matrix(tmp1[y_cols][:-1], y_pred))


/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


[[56696   907]
 [   71   204]]


modeling하여 data를 늘려서 적용해보니 갑자기 10 265인 비율이 71 204로 꺠짐 why? 이유 찾아야함

In [44]:
tmp1[x_cols][:-1]

loc1_tem  loc1_hum  out_hum
0         16.00     24.00    42.00
1         14.00     28.00    59.00
2         13.00     33.00    56.00
3         13.00     33.00    30.00
4         16.00     28.00    20.00
...         ...       ...      ...
58743     10.43     36.02    42.92
58744     10.52     36.39    43.86
58745     10.52     37.09    44.02
58746     10.43     37.18    43.04
58747     10.49     38.00    45.33

[57878 rows x 3 columns]